### ISSUE TITLE
attributesToProps converts case to lower case

### ISSUE THREAD

**Issue by h0jeZvgoxFepBQ2C** [remarkablemark/html-react-parser#263]
It seems that the ```attributesToProps``` method is converting all keys to lowercase (see https://github.com/remarkablemark/html-react-parser/blob/master/lib/attributes-to-props.js#L37 ).

Why is this happening?
We would like to pass case-sensitive attributes to our dynamic html content and with the current implementation it doesn't work, since all attributes are converted to lower case?

**Comment by remarkablemark** [comment]
Thanks for the question @h0jeZvgoxFepBQ2C, `attributesToProps` converts all keys to lowercase because the DOM/SVG attributes get remapped to React props (e.g., `class` to `className`).

On the browser, the attributes technically don't care about case-sensitivity. Is there a particular reason why you need case-sensitive attributes?

There are [htmlparser2 options](https://github.com/remarkablemark/html-react-parser/tree/v1.2.7#htmlparser2), but they only work on Node.js (server-side) and do not work in the browser (client-side).

**Comment by h0jeZvgoxFepBQ2C** [comment]
Thanks for the follow up @remarkablemark 

Actually we would like to pass react specific arguments to our html, so we can use these attributes directly.. F.e. imagine a div should be replaced with a countdown element with the argument ```endsAt```:

```html
<div role="countdown" endsAt="2021-10-10 01:01:01+0200"/>
```
should be converted to
```jsx
<Countdown endsAt="2021-10-10 01:01:01+0200"/>
```

So if the arguments are always downcased, I cannot use them for the react components?

**Comment by h0jeZvgoxFepBQ2C** [comment]
@remarkablemark Any thoughts on this? Your library would offer much more possibilities if it doesn't convert all keys to downcase. And as you said, browsers don't care about case sensitivity, so it wouldn't make a difference? 

Right now there is only this downside, that we can't use it to pass props directly to other React components?

**Comment by remarkablemark** [comment]
@h0jeZvgoxFepBQ2C If you're using the library only on the server-side (Node.js), then you can override the [htmlparser2 options](https://github.com/remarkablemark/html-react-parser/tree/v1.4.7#htmlparser2). You can find a list of the parser options [here](https://github.com/fb55/htmlparser2/wiki/Parser-options#option-xmlmode).

**Comment by h0jeZvgoxFepBQ2C** [comment]
I know, but we are clientside rendering

**Comment by Heilemann** [comment]
Adding my vote to this. If possible it would be nice with an option to enable the client side version to work the same as the server side version. Some use cases, like mine, don't need to care about e.g. XSS issues.

**Comment by h0jeZvgoxFepBQ2C** [comment]
@remarkablemark would you accept a PR for this, to make the option also available for the client side?

**Comment by remarkablemark** [comment]
@h0jeZvgoxFepBQ2C I feel the approach will add more complexity to `html-dom-parser`.

If you really want to achieve this, one approach is to override Webpack to ensure `htmlparser2` is imported in `html-dom-parser`. The downside is your bundle size will increase a lot.

**Comment by T-Fletcher** [comment]
I second this issue, case conversion is a pain as you can't export a string as React components - I thought this would be a key use case but maybe not.

**Comment by remarkablemark** [comment]
@Yankovsky Thanks for the suggestion. I'm just concerned that this may not be backwards compatible and could cause a breaking change for existing users.

Currently, `html-dom-parser` falls back to `new DOMParser().parseFromString(html, 'text/html')`: https://github.com/remarkablemark/html-dom-parser/blob/v5.0.8/src/client/domparser.ts#L37

**Comment by Yankovsky** [comment]
Yeah, I checked the code already. It won't be as easy as just changing the mime type to `text/xml`, but it should be possible. Do you have some tests that we can run the xml parser against?

**Comment by bressler95tops** [comment]
I definitely agree with this.  If you are trying to do this client side on a static website, it's going to cause components to not be recognized by the browser.  I thought well no problem I'll pass options to it, and that doesn't work because html2parser is server side only, which essentially renders this useless on the client side since react wants non internal components to start with uppercase.  I will have to find something else or come up with something hacky sadly
